libs/community/langchain_community/chains/pebblo_retrieval/base.py

```diff
@@ -98,10 +101,20 @@ def _call(
         """
         prompt_time = datetime.datetime.now().isoformat()
         PebbloRetrievalQA.set_prompt_sent(value=False)
-        _run_manager = run_manager or CallbackManagerForChainRun.get_noop_manager()
+        PebbloRetrievalQA.set_prompt_gov_sent(value=False)
         question = inputs[self.input_key]
+
         auth_context = inputs.get(self.auth_context_key, {})
         semantic_context = inputs.get(self.semantic_context_key, {})
+        is_valid_prompt = self._check_prompt_validity(question, semantic_context)
```

Comment:
You may need to call this conditionally(only if `semantic_context.pebblo_semantic_entities.deny` is present)

Reply:
I have made the required changes